From: Changli Gao <xiaosuo@gmail.com>
Date: Sun, 22 Aug 2010 17:25:05 +0000
Subject: header: fix broken headers for user space
Bug-Debian: http://bugs.debian.org/692133

commit 09cd2b99c6cdd1e14e84c1febca2fb91e9f4e5ba upstream.

__packed is only defined in kernel space, so we should use
__attribute__((packed)) for the code shared between kernel and user space.

Two __attribute() annotations are replaced with __attribute__() too.

Signed-off-by: Changli Gao <xiaosuo@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Backported to 2.6.32: most changes were unneeded, as the headers had
 not yet been wrongly converted to use __packed]
---
--- a/include/linux/if_pppox.h
+++ b/include/linux/if_pppox.h
@@ -92,7 +92,7 @@ struct pppoe_tag {
 	__be16 tag_type;
 	__be16 tag_len;
 	char tag_data[0];
-} __attribute ((packed));
+} __attribute__ ((packed));
 
 /* Tag identifiers */
 #define PTT_EOL		__cpu_to_be16(0x0000)
--- a/include/linux/rfkill.h
+++ b/include/linux/rfkill.h
@@ -78,7 +78,7 @@ struct rfkill_event {
 	__u8  type;
 	__u8  op;
 	__u8  soft, hard;
-} __packed;
+} __attribute__((packed));
 
 /*
  * We are planning to be backward and forward compatible with changes
